src/msg/async/AsyncConnection.cc

```diff
@@ -118,6 +118,61 @@ static void alloc_aligned_buffer(bufferlist& data, unsigned len, unsigned off)
   }
 }
 
+ /**
+ * The DelayedDelivery is for injecting delays into Message delivery off
+ * the socket. It is only enabled if delays are requested, and if they
+ * are then it pulls Messages off the DelayQueue and puts them into the
+ * AsyncMessenger event queue.
+ */
+class AsyncConnection::DelayedDelivery : public EventCallback {
```

Comment:
could hide it in an anonymous namespace if it's not supposed to be used out side of this compilation unit.